lib/puppet/application/apply.rb

```diff
@@ -164,11 +167,15 @@ def app_defaults
   end
 
   def run_command
-    if options[:catalog]
-      apply
-    else
-      main
+    lock do
+      if options[:catalog]
+        apply
+      else
+        main
+      end
     end
+  rescue Puppet::LockError
+    Puppet.notice "Run of #{client_class} already in progress; skipping  (#{lockfile_path} exists)"
```

Comment:
Oops, this isn't defined in `apply`. Caught by an acceptance run at https://jenkins.puppetlabs.com/job/platform_puppet_intn-van-sys_master/24/SLAVE_LABEL=beaker,TEST_TARGET=redhat7-64a/console.

Which means our acceptance runs have multiple puppet apply happening or are failing to cleanup the lock file. That's interesting.